src/Perception/src/ArucoDetector.cpp

```diff
@@ -0,0 +1,300 @@
+                                                                                         {"7X7_250", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_7X7_250},
+                                                                                         {"7X7_1000",cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_7X7_1000},
+                                                                                         {"ARUCO_ORIGINAL",cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_ARUCO_ORIGINAL} };
+};
+
+ArucoDetector::ArucoDetector() :
+               m_pimpl(std::make_unique<Impl>())
+{
+}
+
+ArucoDetector::~ArucoDetector() = default;
+
+bool ArucoDetector::initialize(std::weak_ptr<IParametersHandler> handler)
+{
+    std::string_view printPrefix = "[ArucoDetector::initialize] ";
+    auto handle = handler.lock();
+    if (handle == nullptr)
+    {
+        log()->error("{} "
+        "The parameter handler has expired. Please check its scope.", printPrefix);
+        return false;
+    }
+
+    std::string dictName;
+    if (!handle->getParameter("marker_dictionary", dictName))
+    {
+        log()->error("{} "
+        "The parameter handler could not find \" marker_dictionary \" in the configuration file.",
+        printPrefix);
+        return false;
+    }
+
+    if (m_pimpl->availableDict.find(dictName) == m_pimpl->availableDict.end())
+    {
+        std::cerr << printPrefix
+        << "Undefined value set to \" marker_dictionary \" in the configuration file. Please choose one among \n"
+        << "available options: \"4X4_50\", \"4X4_100\", \"4X4_250\", \"4X4_1000\", \n"
+        << "\"5X5_50\", \"5X5_100\", \"5X5_250\", \"5X5_1000\", \n"
+        << "\"6X6_50\", \"6X6_100\", \"6X6_250\", \"6X6_1000\", \n"
+        << "\"7X7_50\", \"7X7_100\", \"7X7_250\", \"7X7_1000\", \n"
+        << " \"ARUCO_ORIGINAL\", \n"
+        << "options coherent with v3.4.0 in https://docs.opencv.org/3.4.1/d9/d6a/group__aruco.html"
+        << std::endl;
+        return false;
+    }
+
+    m_pimpl->dictionary = cv::aruco::getPredefinedDictionary(m_pimpl->availableDict.at(dictName));
+
+    if (!handle->getParameter("marker_length", m_pimpl->markerLength))
+    {
+        log()->error("{} "
+        "The parameter handler could not find \" marker_length \" in the configuration file.",
+        printPrefix);
+        return false;
+    }
+
+    std::vector<double> calibVec;
+    if (!handle->getParameter("camera_matrix",
+                               GenericContainer::make_vector(calibVec, GenericContainer::VectorResizeMode::Resizable)))
+    {
+        log()->error("{} "
+        "The parameter handler could not find \" camera_matrix \" in the configuration file.",
+        printPrefix);
+        return false;
+    }
+
+    m_pimpl->cameraMatrix = cv::Mat(3, 3, CV_64F);
+    std::memcpy(m_pimpl->cameraMatrix.data, calibVec.data(), calibVec.size()*sizeof(double));
```

Comment:
I think you can use an ``Eigen::VectorXd`` instead of a ``std::vector`` for ``calibVec``. In this way, you can avoid the ``memcpy`` (that is a bit risky) and use the eigen assignment methods instead (eventually using maps).

The same holds below.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/159/commits/daa331009be38c957e5c29b015065383907aa6e7, thanks!